### PR TITLE
feat(dataframe): implement "DataFrame._pylist_to_pyarrow"

### PR DESCRIPTION
**Pull request by AChenQ**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-28v2czn 基于 PagingList 实现 DataFrame.__init__ 方法](https://app.clickup.com/t/28v2czn)

File: graviti/dataframe/frame.py [hunk 1]
```diff
@@ -598,6 +600,93 @@ def extend(self, values: Union[Iterable[Dict[str, Any]], "DataFrame"]) -> None:
...
+
+                if isinstance(field, pt.record):
+                    process_func[name], sub_need_process = DataFrame._process_record(
+                        item, field.to_builtin()
+                    )
+                    need_process = need_process or sub_need_process
+                elif isinstance(field, pt.array):
+                    process_func[name], sub_need_process = DataFrame._process_array(item, field)
+                    need_process = need_process or sub_need_process
+                elif field.container == FileSeries:
```
Comment (linjiX) [inline graviti/dataframe/frame.py:661]: should we always check `container`？
```python
field.container == DataFrame
field.container == ArraySeries
```

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataframe/frame.py [hunk 2]
```diff
@@ -598,6 +600,93 @@ def extend(self, values: Union[Iterable[Dict[str, Any]], "DataFrame"]) -> None:
...
+
+                process_func, need_process = DataFrame._process_array(value, schema.items)
+                return lambda x: list(map(process_func, x)), need_process
+
+        elif schema.items.container == FileSeries:
+            for value in values:
+                if value is None:
+                    continue
+
+                process_func, need_process = DataFrame._process_file(value)
```
Comment (linjiX) [inline graviti/dataframe/frame.py:626]: `process_func` -> `processor` or `processors`

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataframe/frame.py [hunk 3]
```diff
@@ -598,6 +600,100 @@ def extend(self, values: Union[Iterable[Dict[str, Any]], "DataFrame"]) -> None:
...
+        for row in values:
+            for name, field in schema.fields.items():
+                if name in processors:
+                    continue
+
+                item = row[name]
+                if item is None:
+                    continue
+
+                if field.container == DataFrame:
```
Comment (linjiX) [inline graviti/dataframe/frame.py:658]: set container as a local variable?
```python
container = field.container
```

**Review by linjiX — CHANGES_REQUESTED** [review]

**Review by linjiX — APPROVED** [review]

**Pull request merged**